File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/readmodel/ConferenceQueryService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using Conference.Common;
using ECommon.Components;
using ECommon.Dapper;

namespace ConferenceManagement.ReadModel
{
    [Component]
    public class ConferenceQueryService
    {
        public ConferenceDTO FindConference(string slug)
        {
            using (var connection = GetConnection())
            {
                return connection.QueryList<ConferenceDTO>(new { Slug = slug }, ConfigSettings.ConferenceTable).SingleOrDefault();
            }
        }
        public ConferenceDTO FindConference(string email, string accessCode)
        {
            using (var connection = GetConnection())
            {
                return connection.QueryList<ConferenceDTO>(new { OwnerEmail = email, AccessCode = accessCode }, ConfigSettings.ConferenceTable).SingleOrDefault();
            }
        }
        public IEnumerable<SeatTypeDTO> FindSeatTypes(Guid conferenceId)
        {
            using (var connection = GetConnection())
            {
                return connection.QueryList<SeatTypeDTO>(new { ConferenceId = conferenceId }, ConfigSettings.SeatTypeTable).ToList();
            }
        }
        public SeatTypeDTO FindSeatType(Guid seatTypeId)
        {
            using (var connection = GetConnection())
            {
                return connection.QueryList<SeatTypeDTO>(new { Id = seatTypeId }, ConfigSettings.SeatTypeTable).SingleOrDefault();
            }
        }
        public IEnumerable<OrderDTO> FindOrders(Guid conferenceId)
        {
            using (var connection = GetConnection())
            {
                var orders = connection.QueryList<OrderDTO>(new { ConferenceId = conferenceId }, ConfigSettings.OrderTable);
                foreach (var order in orders)
                {
                    order.SetAttendees(connection.QueryList<AttendeeDTO>(new { OrderId = order.OrderId }, ConfigSettings.OrderSeatAssignmentsTable).ToList());
                }
                return orders;
            }
        }

        private IDbConnection GetConnection()
        {
            return new SqlConnection(ConfigSettings.ConferenceConnectionString);
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/readmodel/ConferenceQueryService.java
package com.microsoft.conference.management.readmodel;

import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
import com.microsoft.conference.common.dataobject.ConferenceDO;
import com.microsoft.conference.common.dataobject.OrderDO;
import com.microsoft.conference.common.dataobject.OrderSeatAssignmentDO;
import com.microsoft.conference.common.dataobject.SeatTypeDO;
import com.microsoft.conference.common.mapper.ConferenceMapper;
import com.microsoft.conference.common.mapper.OrderMapper;
import com.microsoft.conference.common.mapper.OrderSeatAssignmentMapper;
import com.microsoft.conference.common.mapper.SeatTypeMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Service
public class ConferenceQueryService {

    @Autowired
    private ConferenceMapper conferenceMapper;
    @Autowired
    private SeatTypeMapper seatTypeMapper;
    @Autowired
    private OrderMapper orderMapper;
    @Autowired
    private OrderSeatAssignmentMapper orderSeatAssignmentMapper;

    public ConferenceVO findConference(String slug) {
        LambdaQueryWrapper<ConferenceDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(ConferenceDO::getSlug, slug);
        ConferenceDO conferenceDO = conferenceMapper.selectOne(queryWrapper);
        return DTOExtensions.INSTANCE.toVO(conferenceDO);
    }

    public ConferenceVO findConference(String email, String accessCode) {
        LambdaQueryWrapper<ConferenceDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(ConferenceDO::getOwnerEmail, email);
        queryWrapper.eq(ConferenceDO::getAccessCode, accessCode);
        ConferenceDO conferenceDO = conferenceMapper.selectOne(queryWrapper);
        return DTOExtensions.INSTANCE.toVO(conferenceDO);
    }

    public List<SeatTypeVO> findSeatTypes(String conferenceId) {
        LambdaQueryWrapper<SeatTypeDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(SeatTypeDO::getConferenceId, conferenceId);
        List<SeatTypeDO> seatTypeDOS = seatTypeMapper.selectList(queryWrapper);
        return Optional.ofNullable(seatTypeDOS).orElse(new ArrayList<>())
                .stream().map(DTOExtensions.INSTANCE::toVO)
                .collect(Collectors.toList());
    }

    public SeatTypeVO findSeatType(String seatTypeId) {
        LambdaQueryWrapper<SeatTypeDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(SeatTypeDO::getSeatTypeId, seatTypeId);
        SeatTypeDO seatTypeDO = seatTypeMapper.selectOne(queryWrapper);
        return DTOExtensions.INSTANCE.toVO(seatTypeDO);
    }

    public List<OrderVO> findOrders(String conferenceId) {
        LambdaQueryWrapper<OrderDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(OrderDO::getConferenceId, conferenceId);
        List<OrderDO> orderDOS = orderMapper.selectList(queryWrapper);
        return Optional.ofNullable(orderDOS).orElse(new ArrayList<>()).stream().map(orderDO -> {
            LambdaQueryWrapper<OrderSeatAssignmentDO> wrapper = new LambdaQueryWrapper<>();
            List<OrderSeatAssignmentDO> seatAssignmentDOS = orderSeatAssignmentMapper.selectList(wrapper);
            OrderVO orderVO = DTOExtensions.INSTANCE.toVO(orderDO);
            orderVO.setAttendees(Optional.ofNullable(seatAssignmentDOS).orElse(new ArrayList<>()).stream()
                    .map(DTOExtensions.INSTANCE::toVO)
                    .collect(Collectors.toList()));
            return orderVO;
        }).collect(Collectors.toList());
    }
}
